src/store/modules/chatMessages.js

```diff
@@ -1,123 +1,49 @@
-// Don't mind this import. I will use it in my next PR.
-// import ChatMessageService from '../../services/ChatMessage/ChatMessage';
+import ChatMessageService from "../../services/chatMessages/chatMessages";
 
-const state = [
-  {
-    chatId: 1,
-    senderName: "Virtual Concierge",
-    senderIcon: "logo.png",
-    messages: [
-      {
-        chatMessageId: 1,
-        receivedTime: new Date("August 11, 1975 23:15:30"),
-        isUser: false,
-        text: "Hi, how can I help you?",
-      },
-      {
-        chatMessageId: 2,
-        receivedTime: new Date("August 13, 1975 23:15:30"),
-        isUser: true,
-        text: "I need help! My desk is on fire!",
-      },
-      {
-        chatMessageId: 3,
-        receivedTime: new Date("August 15, 1975 23:15:30"),
-        isUser: false,
-        text: "I appreciate your concern. Apply water vigorously and call 911.",
-      },
-      {
-        chatMessageId: 4,
-        receivedTime: new Date("August 17, 1975 23:15:30"),
-        isUser: true,
-        text: "I meant I am working too hard; I am all fired up right now.",
-      },
-      {
-        chatMessageId: 5,
-        receivedTime: new Date("August 19, 1975 23:15:30"),
-        isUser: false,
-        text: "Oh... Well, this is akward.",
-      },
-      {
-        chatMessageId: 6,
-        receivedTime: new Date("August 20, 1975 23:15:30"),
-        isUser: false,
-        text: "Hi, how can I help you?",
-      },
-      {
-        chatMessageId: 7,
-        receivedTime: new Date("August 22, 1975 23:15:30"),
-        isUser: true,
-        text: "I need help! My desk is on fire!",
-      },
-      {
-        chatMessageId: 8,
-        receivedTime: new Date("August 23, 1975 23:15:30"),
-        isUser: false,
-        text: "I appreciate your concern. Apply water vigorously and call 911.",
-      },
-      {
-        chatMessageId: 9,
-        receivedTime: new Date("August 25, 1975 23:15:30"),
-        isUser: true,
-        text: "I meant I am working too hard; I am all fired up right now.",
-      },
-      {
-        chatMessageId: 10,
-        receivedTime: new Date("August 27, 1975 23:15:30"),
-        isUser: false,
-        text: "Oh... Well, this is akward.",
-      },
-    ],
-  },
-];
+const state = {
+  chatMessages: [],
+};
 
 // getters
 const getters = {
   // Get virtual concierge items
   getChatMessagesOrderedByDate(state) {
     // state[0] will change to the selected inbox item
-    const conversation = [...state[0].messages].sort((a, b) => {
-      if (a.receivedTime > b.receivedTime) {
-        return -1;
-      }
-      if (a.receivedTime < b.receivedTime) {
-        return 1;
-      }
-      return 0;
-    });
-    return conversation;
+    if (state.chatMessages.length !== 0) {
+      const conversation = [...state.chatMessages[0].messages].sort((a, b) => {
```

Comment:
You could use state.chatMessages[0].messages here, since messages is an array.

Reply:
To add onto this we can specify which chatMessages to get now from inbox/selectedInboxItemId. We may need to coordinate this change

Reply:
Taking a look now.

Reply:
I actually can't do what is suggested because that would change the state. A getter cannot mutate the state.

Reply:
Ohh, that makes sense. In that case, could you add a comment to clarify that ? It's not very obvious at first glance that this is a quick way to get a copy of the array.

Reply:
Sure!